### PR TITLE
[node] Refactor 'globals.d.ts'

### PR DESCRIPTION
**Pull request by Semigradsky**
Make `globals.d.ts` more easy for reading and editing.

- Move 'timers'-specific types to `timers.d.ts`
- Move 'buffer'-specific types to `buffer.d.ts`
- Move 'errors'-specific types to `errors.d.ts` (new file)
- Move 'stream'-specific types to `stream.d.ts`
- Add `URL` and `URLSearchParams` globals in `url.d.ts`
- Add `TextDecoder` and `TextEncoder` globals in `util.d.ts`
- Make `global.gc` optional
- Add himself to `index.d.ts` for receiving notifications about new `node` issues

### CODE REVIEW

File: types/node/errors.d.ts [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    interface Global {
+        Error: typeof Error;
+        EvalError: typeof EvalError;
+        RangeError: typeof RangeError;
+        ReferenceError: typeof ReferenceError;
+        SyntaxError: typeof SyntaxError;
+        TypeError: typeof TypeError;
+        URIError: typeof URIError;
+    }
+}
```
Comment (ExE-Boss) [inline types/node/errors.d.ts:132]: There is no built‑in `"errors"` module in **NodeJS**, so this should either be kept in `globals.d.ts`, or renamed to `globals.errors.d.ts`.

**Review by ExE-Boss — CHANGES_REQUESTED** [review]

File: types/node/buffer.d.ts [hunk 2]
```diff
@@ -19,4 +19,257 @@ declare module "buffer" {
...
+    writeDoubleBE(value: number, offset?: number): number;
+
+    fill(value: string | Uint8Array | number, offset?: number, end?: number, encoding?: BufferEncoding): this;
+
+    indexOf(value: string | number | Uint8Array, byteOffset?: number, encoding?: BufferEncoding): number;
+    lastIndexOf(value: string | number | Uint8Array, byteOffset?: number, encoding?: BufferEncoding): number;
+    entries(): IterableIterator<[number, number]>;
+    includes(value: string | number | Buffer, byteOffset?: number, encoding?: BufferEncoding): boolean;
+    keys(): IterableIterator<number>;
+    values(): IterableIterator<number>;
```
Comment (ExE-Boss) [inline types/node/buffer.d.ts:274]: This should be moved into the `global` block inside `module "buffer"`, like in <https://github.com/DefinitelyTyped/DefinitelyTyped/pull/45845>.

File: types/node/timers.d.ts [hunk 3]
```diff
@@ -13,4 +13,54 @@ declare module "timers" {
...
+}
+declare function clearTimeout(timeoutId: NodeJS.Timeout): void;
+declare function setInterval(callback: (...args: any[]) => void, ms: number, ...args: any[]): NodeJS.Timeout;
+declare function clearInterval(intervalId: NodeJS.Timeout): void;
+declare function setImmediate(callback: (...args: any[]) => void, ...args: any[]): NodeJS.Immediate;
+declare namespace setImmediate {
+    function __promisify__(): Promise<void>;
+    function __promisify__<T>(value: T): Promise<T>;
 }
+declare function clearImmediate(immediateId: NodeJS.Immediate): void;
```
Comment (ExE-Boss) [inline types/node/timers.d.ts:66]: This should be moved into the `global` block inside `module "timers"`.

**Review by ExE-Boss — CHANGES_REQUESTED** [review]

File: types/node/buffer.d.ts (hunk 2 above)
Comment (Semigradsky) [reply types/node/buffer.d.ts:274]: In this case I have the error 🤔 :
```
> dtslint types "node"

testing from 3.2 to 3.4 in /home/dmitry/H/DefinitelyTyped/types/node/ts3.4
Error: Errors in typescript@3.4 for external dependencies:
../buffer.d.ts(128,41): error TS2694: Namespace 'global.NodeJS' has no exported member 'ArrayBufferView'.
```

127-130 lines:
```ts
            static byteLength(
                string: string | NodeJS.ArrayBufferView | ArrayBuffer | SharedArrayBuffer,
                encoding?: BufferEncoding
            ): number;
```

File: types/node/errors.d.ts (hunk 1 above)
Comment (Semigradsky) [reply types/node/errors.d.ts]: Ok, I have renamed the file.

File: types/node/timers.d.ts (hunk 3 above)
Comment (Semigradsky) [reply types/node/timers.d.ts]: Done

File: types/node/buffer.d.ts (hunk 2 above)
Comment (Semigradsky) [reply types/node/buffer.d.ts:274]: @ExE-Boss hi! Is it really needed change?

Comment (Semigradsky) [reply types/node/buffer.d.ts]: @ExE-Boss what about moving exposing to global to separate `MODULE_NAME.global.d.ts` files? See 16a2ec94ba1f48c0e9f33d4ab870ec064fbe2473 In this case I haven't previous error.

**Pull request closed without merging**